File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/VoiceActivity.java
package ch.rmy.android.http_shortcuts;

import android.app.SearchManager;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.widget.Toast;

import ch.rmy.android.http_shortcuts.realm.Controller;
import ch.rmy.android.http_shortcuts.realm.models.Shortcut;
import ch.rmy.android.http_shortcuts.utils.IntentUtil;

// THIS IMPLEMENTATION IS EXPERIMENTAL ONLY
public class VoiceActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        String shortcutName = getIntent().getStringExtra(SearchManager.QUERY);
        if (shortcutName == null) {
            return;
        }

        Controller controller = destroyer.own(new Controller(getContext()));
        Shortcut shortcut = controller.getShortcutByName(shortcutName);
        if (shortcut == null) {
            Toast.makeText(getContext(), "Shortcut \"" + shortcutName + "\" not found", Toast.LENGTH_LONG).show();
            finishWithoutAnimation();
            return;
        }

        Intent intent = IntentUtil.createIntent(getContext(), shortcut.getId());
        startActivity(intent);
        finishWithoutAnimation();
    }

}

File: HTTPShortcuts/app/src/main/kotlin/ch/rmy/android/http_shortcuts/activities/VoiceActivity.kt
package ch.rmy.android.http_shortcuts.activities

import android.app.SearchManager
import android.os.Bundle
import ch.rmy.android.http_shortcuts.data.Controller
import ch.rmy.android.http_shortcuts.extensions.showToast
import ch.rmy.android.http_shortcuts.extensions.startActivity

// THIS IMPLEMENTATION IS EXPERIMENTAL ONLY
class VoiceActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val shortcutName = intent.getStringExtra(SearchManager.QUERY) ?: return

        val controller = destroyer.own(Controller())
        val shortcut = controller.getShortcutByName(shortcutName) ?: run {
            showToast("Shortcut \"$shortcutName\" not found")
            finishWithoutAnimation()
            return
        }

        ExecuteActivity.IntentBuilder(context, shortcut.id)
            .build()
            .startActivity(this)
        finishWithoutAnimation()
    }

}
